src/main/java/org/opensearch/agent/tools/VisualizationsTool.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.agent.tools;
+
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+
+import org.opensearch.ExceptionsHelper;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.client.Client;
+import org.opensearch.client.Requests;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.index.IndexNotFoundException;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.ml.common.spi.tools.Tool;
+import org.opensearch.ml.common.spi.tools.ToolAnnotation;
+import org.opensearch.search.SearchHits;
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+@ToolAnnotation(VisualizationsTool.TYPE)
+public class VisualizationsTool implements Tool {
+    public static final String NAME = "FindVisualizations";
+    public static final String TYPE = "VisualizationTool";
+    public static final String VERSION = "v1.0";
+
+    public static final String SAVED_OBJECT_TYPE = "visualization";
+    private static final String DEFAULT_DESCRIPTION =
+        "Use this tool to find user created visualizations. This tool takes the visualization name as input and returns the first 3 matching visualizations";
```

Comment:
Does `3` needs to be static? Can't we do like 3 by default but the input is dynamic?

Reply:
good suggestion, update the number to dynamic setting.

Reply:
May be update the description as well?

Reply:
the specific number in description has been removed. now it's `"Use this tool to find user created visualizations. This tool takes the visualization name as input and returns the matching visualizations"
`